TundraTD/Assets/Scripts/Spells/SpellClasses/FireballSpell.cs

```diff
@@ -46,17 +50,46 @@ public class FireballSpell : MagicSpell
 
         public override void ExecuteSpell()
         {
-            Debug.Log($"Meteor cast. Hit damage: {HitDamageValue}; Burn duration: {BurnDuration}; Burn damage: {BurnDamage}; Slowness duration: {SlownessDuration}.");
-            throw new NotImplementedException();
+            Debug.Log("Fireball has been executed!");
+            if (Physics.Raycast(Camera.main.transform.position, Camera.main.transform.forward, out var hit))
+            {
+                target = hit.point;
+                var reflect = Vector3.Reflect(Quaternion.Euler(0, -90, 0) * Camera.main.transform.forward, hit.normal).normalized;
+                transform.position = hit.point + reflect * flyDistance;
+                Debug.DrawLine(transform.position, target, Color.red, 2);
+                Debug.DrawRay(target, Vector3.up * HitDamageRadius, Color.blue, 2);
+            }
         }
 
         private void Update()
         {
+            // Performs flight towards target.
```

Comment:
nit: We could extract this code into a function and call it FlightTowardsTarget

Reply:
Well, that's just one line so I think it's not important